server/src/main/java/org/apache/iotdb/db/http/router/Router.java

```diff
@@ -0,0 +1,109 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iotdb.db.http.router;
+
+import com.google.gson.*;
+import com.librato.metrics.client.Json;
+import io.netty.handler.codec.http.HttpMethod;
+import io.netty.handler.codec.http.QueryStringDecoder;
+import java.io.IOException;
+import java.sql.SQLException;
+import org.apache.iotdb.db.auth.AuthException;
+import org.apache.iotdb.db.exception.StorageEngineException;
+import org.apache.iotdb.db.exception.UnsupportedHttpMethod;
+import org.apache.iotdb.db.exception.metadata.MetadataException;
+import org.apache.iotdb.db.exception.query.QueryProcessException;
+import org.apache.iotdb.db.http.constant.HttpConstant;
+import org.apache.iotdb.db.http.handler.DeleteStorageGroupsHandler;
+import org.apache.iotdb.db.http.handler.DeleteTimeSeriesHandler;
+import org.apache.iotdb.db.http.handler.GetTimeSeriesHandler;
+import org.apache.iotdb.db.http.handler.InsertHandler;
+import org.apache.iotdb.db.http.handler.QueryHandler;
+import org.apache.iotdb.db.http.handler.StorageGroupsHandlers;
+import org.apache.iotdb.db.http.handler.TimeSeriesHandler;
+import org.apache.iotdb.db.http.handler.UsersHandler;
+import org.apache.iotdb.db.http.utils.URIUtils;
+import org.apache.iotdb.tsfile.exception.filter.QueryFilterOptimizationException;
+import org.apache.thrift.TException;
+
+/**
+ * Router that contains information about both route matching and return json.
+ */
+public class Router {
+
+  /**
+   * matching url and return JSON
+   * @param method http method
+   * @param uri uri will be matched
+   * @param json request JSON Object
+   * @return JSON object, may be a JSONArray or JSONObject
+   */
+  public JsonElement route(HttpMethod method, String uri, JsonElement json)
+      throws AuthException, MetadataException, QueryProcessException
+      , StorageEngineException, UnsupportedHttpMethod, SQLException, InterruptedException, QueryFilterOptimizationException, IOException, TException {
+    QueryStringDecoder decoder = new QueryStringDecoder(uri);
+    uri = URIUtils.removeParameter(uri);
+    switch (uri) {
+      case HttpConstant.ROUTING_STORAGE_GROUPS:
+        StorageGroupsHandlers storageGroupsHandlers = new StorageGroupsHandlers();
+        return storageGroupsHandlers.handle(method, json);
+      case HttpConstant.ROUTING_TIME_SERIES:
+        TimeSeriesHandler timeSeriesHandler = new TimeSeriesHandler();
+        return timeSeriesHandler.handle(method, json);
+      case HttpConstant.ROUTING_USER_LOGIN:
+        if (UsersHandler.userLogin(decoder.parameters())) {
+          JsonObject result = new JsonObject();
+          result.addProperty(HttpConstant.RESULT, HttpConstant.SUCCESSFUL_OPERATION);
+          return result;
```

Comment:
Successful JsonObject is used many places. Try to extract it to a method and use
` return getSuccessfulObject();`